BSP/Simulator/Src/BSP_UART.c

```diff
@@ -31,6 +31,12 @@ void BSP_UART_Init(void) {
     setvbuf(stdin, NULL, _IONBF, 0);
     setvbuf(stdout, NULL, _IONBF, 0);
 
+    err = pthread_mutex_init(&rx_mutex, NULL);
```

Comment:
The CLI/BLE PR also edits this file. If we want to merge that one after simulator we can. Just to let you know though.

Reply:
Tank for letting us know. We'll merge after simulator is done